homeassistant/components/lawn_mower/__init__.py

```diff
@@ -0,0 +1,187 @@
+        SERVICE_ENABLE_SCHEDULE, {}, "async_service_enable_schedule"
+    )
+    component.async_register_entity_service(
+        SERVICE_DISABLE_SCHEDULE, {}, "async_service_disable_schedule"
+    )
+    component.async_register_entity_service(SERVICE_DOCK, {}, "async_service_dock")
+
+    return True
+
+
+class Activity(IntFlag):
+    """Activity state of the lawn mower entity."""
+
+    ERROR = 1
+    PAUSED = 2
+    MOWING = 4
+    DOCKED_SCHEDULE_DISABLED = 8
+    DOCKED_SCHEDULE_ENABLED = 16
+
+
+class LawnMowerEntityFeature(IntFlag):
+    """Supported features of the lawn mower entity."""
+
+    START_MOWING = 1
+    PAUSE = 2
+    DOCK = 4
+    ENABLE_SCHEDULE = 8
+    DISABLE_SCHEDULE = 16
+    BATTERY = 32
+    LINK = 64
+
+
+@dataclass
+class LawnMowerEntityEntityDescription(EntityDescription):
+    """A class that describes lawn mower entities."""
+
+
+class LawnMowerEntity(Entity):
+    """Base class for lawn mower entities."""
+
+    entity_description: LawnMowerEntityEntityDescription
+    _attr_activity: Activity | None = None
+    _attr_supported_features: int
+
+    @final
+    @property
+    def state(self) -> Activity | None:
+        """Return the current state."""
+        return self._attr_activity
+
+    @property
+    def supported_features(self) -> int:
+        """Flag lawn mower features that are supported."""
+        return self._attr_supported_features
+
+    def start_mowing(self) -> None:
+        """Start mowing."""
+        raise NotImplementedError()
+
+    async def async_start_mowing(self, **kwargs: Any) -> None:
+        """Start mowing."""
+        await self.hass.async_add_executor_job(partial(self.start_mowing, **kwargs))
+
+    def dock(self) -> None:
+        """Dock the mower."""
+        raise NotImplementedError()
+
+    async def async_dock(self, **kwargs: Any) -> None:
+        """Dock the mower."""
+        await self.hass.async_add_executor_job(partial(self.dock, **kwargs))
+
+    def pause(self) -> None:
+        """Pause the lawn mower."""
+        raise NotImplementedError()
+
+    async def async_pause(self, **kwargs: Any) -> None:
+        """Pause the lawn mower."""
+        await self.hass.async_add_executor_job(partial(self.pause, **kwargs))
+
+    def enable_schedule(self) -> None:
+        """Enable the schedule for the lawn mower."""
+        raise NotImplementedError()
+
+    async def async_enable_schedule(self, **kwargs: Any) -> None:
+        """Enable the schedule for the lawn mower."""
+        await self.hass.async_add_executor_job(partial(self.enable_schedule, **kwargs))
+
+    def disable_schedule(self) -> None:
+        """Disable the schedule for the lawn mower."""
+        raise NotImplementedError()
+
+    async def async_disable_schedule(self, **kwargs: Any) -> None:
+        """Disable the schedule for the lawn mower."""
+        await self.hass.async_add_executor_job(partial(self.disable_schedule, **kwargs))
+
+
+async def async_service_start_mowing(
```

Comment:
That would leave only the enable schedule yes?

Reply:
That would remove all right? Since we aren't doing any validation, how does this affect registering services?

Reply:
When we pass a string to the entity component service helper `async_register_entity_service` it will look up the corresponding method name on the entity and use that method as service handler.

Reply:
sorry for the text spam, figured it out. I think.